app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -36,21 +32,41 @@ def customer
       @customer ||= find_or_create_customer
     end
 
+    def update_billing_address(billing_address)
```

Comment:
I think it's quite confusing to have `update_billing_address`, `update_billing_details` and `update_and_save_billing_details` methods, as you need to jump between them to understand what is done there.

At least the last two need to be merged, IMO.